server/handlers/order.js

```diff
@@ -0,0 +1,63 @@
+var dao = require("../dao/dao");
+const { Order } = require("../models/order");
+const { validationResult } = require("express-validator");
+const {
+  clientIDPathValidator,
+  productIDPathValidator,
+  productQtyPathValidator,
+  productsValidator,
+} = require("./shared_validators");
+
+exports.createOrderValidatorChain = [
+  clientIDPathValidator,
+  productsValidator,
+  productIDPathValidator,
+  productQtyPathValidator,
+];
+
+exports.createOrderHandler = async function (req, res, next) {
+  // Insert the new order
+
+  // productPrice is hardcoded to 1 as a tempoarary solution for now, will be fixed in the next sprints
+
+  var result;
+  var totalPrice = 0;
+  var productPrice = 1;
+
+  var isNegativeQuantity = false;
+  req.body.products.forEach((product) => {
+    totalPrice = totalPrice + product.quantity * productPrice;
+    if (product.quantity < 0) {
+      isNegativeQuantity = true;
+    }
+  });
+
+  if (isNegativeQuantity) {
+    console.error(`Negative Quantity is found: ${err}`);
+    return res.status(400).end();
+  }
+  var obj = {
+    clientId: req.body.clientId.toString(),
+    products: JSON.stringify(req.body.products),
+    status: "WAITING",
+    totalPrice: totalPrice.toString(),
+  };
+  try {
+    result = await dao.createOrder(
+      obj.clientId,
+      obj.products,
+      obj.status,
+      obj.totalPrice
+    );
+  } catch (err) {
+    console.error(`CreateOrder() -> couldn't create order: ${err}`);
+    return res.status(500).end();
+  }
+
+  if (!result) {
+    console.error(`CreateOrder() -> couldn't retrieve newly created order`);
+    return res.status(404).end();
+  }
+
+  res.json(Order.fromMongoJSON(obj));
+};
```

Comment:
When returning, products has `\` all over
`{
  "clientId": "318d971d89d6240eb03742d7",
  "products": "[{\"productId\":\"718d971d89d6240eb03742d7\",\"quantity\":3},{\"productId\":\"298d971d89d6240eb03742d7\",\"quantity\":4}]",
  "status": "WAITING",
  "totalPrice": "7"
}`

this is the result of the example provided.
It should also return the id of the order